src/prediction.jl

```diff
@@ -38,6 +38,19 @@ function predict{T<:BlasReal}(pm::LinearPred, θ::StridedVector{T}, X::StridedMa
     X'θ
 end
 
+function predict!{T<:BlasReal}(r::StridedVector{T}, pm::LinearPred, θ::StridedVector{T}, X::StridedMatrix{T})
+    d = pm.dim
+    n = size(X,2)
+    @_checkdims length(θ) == size(X,1) == pm.dim && n == length(r)
+    for col = 1:n
+        @inbounds r[col] = 0
+        for row = 1:d
+            @inbounds r[col] += X[row, col] * θ[row]
```

Comment:
Should use ``A_mul_B!``, ``At_mul_B!`` etc, instead of using a for-loop to do matrix multiplication, which is way slower than the BLAS routines! The same apply to all the following methods.